@here/harp-omv-datasource/test/MapViewPickingTest.ts

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// tslint:disable:no-unused-expression
+//    expect-type assertions are unused expressions and are perfectly valid
+
+// tslint:disable:no-empty
+//    lots of stubs are needed which are just placeholders and are empty
+
+// tslint:disable:only-arrow-functions
+//    Mocha discourages using arrow functions, see https://mochajs.org/#arrow-functions
+
+import {
+    DecodedTile,
+    ExtendedTileInfo,
+    Geometry,
+    GeometryType
+} from "@here/harp-datasource-protocol";
+import { GeoJsonDataProvider } from "@here/harp-geojson-datasource";
+import { GeoCoordinates, TileKey } from "@here/harp-geoutils";
+import { MapView, MapViewEventNames, ThemeLoader, TileLoaderState } from "@here/harp-mapview";
+import { GeoJsonTiler } from "@here/harp-mapview-decoder/lib/GeoJsonTiler";
+import { waitForEvent } from "@here/harp-test-utils/";
+import { loadTestResource } from "@here/harp-test-utils/lib/TestDataUtils";
+import * as TestUtils from "@here/harp-test-utils/lib/WebGLStub";
+import { FontCatalog } from "@here/harp-text-canvas";
+import { GlyphTextureCache } from "@here/harp-text-canvas/lib/rendering/GlyphTextureCache";
+import { assert } from "chai";
+import * as sinon from "sinon";
+import * as THREE from "three";
+import { OmvTileDecoder } from "../index-worker";
+import { OmvDataSource } from "../lib/OmvDataSource";
+import { OmvTile } from "../lib/OmvTile";
+import { GEOJSON_DATA, THEME } from "./resources/geoJsonData";
+
+declare const global: any;
+
+// repeat a guard function until guard return true or repeat count <= 0
```

Comment:
It's very useful, please add it to `@here/harp-test-utils`.

This may be integrated with `lastWaiterError` defined there, so in case of `timeouts` you can display something useful like this:

"Failed waiting until `${fn.toString}` is true."

BTW, i would remove `repeat`
This is something like hardcoded timeout, and for this we have `this.timeout()` in mocha.

Reply:
I must leave the repeat counter because of node v8  event loop.
While some interval events exist in the node event loop, the node process won't stop. I need to clear the intervals to normally exit from the process.

Reply:
So this, is only hiding problem under carpet.
The real problem is that your code still works after mocha decides to `timeout` and abort test ...
Again, look at `willEventually` how it's done - there is `afterEach` handler added, that ensures that `willEventually` will abort if mocha done with current test.

BTW, Your use case can be implemented with
```
await willEventually(() => {
        assert.isTrue(mapView.visibleTileSet.allVisibleTilesLoaded);
});
```
so 1) you reuse code, 2) you have clear info when you hit timeout due to condition never ocurred 3) gain proper cleanup (already worked out in pain for `willEventually` and friends).

Reply:
Done.